doc/extensibility/IProjectItemCotnextMenuProvider.md

```diff
@@ -0,0 +1,80 @@
+# IProjectContextMenuProvider
+
+Extension point for suppling the context menu group `Guid` and the menu Id `int` to display
+to the user. This interface does not let you _implement_ new context menus to show to the
+user, but only lets yous elect what menu to show based on the `IProjectTree` selected.
+See [Extending Menus and Commands](https://msdn.microsoft.com/en-us/library/bb165937.aspx)
+for information on how to implement context menus.
+
+## Extension Point
+
+* Contract: `[Export(typeof(IProjectItemContextMenuProvider))]`
+* Metadata: No special metadata. Uses `[Order]` and `[AppliesTo]`
+
+## Interface
+
+``` CSharp
+/// <summary>
+/// An extension component to provide context menu to be shown in the solution explorer for a specific project item.
+/// </summary>
+[ProjectSystemContract(ProjectSystemContractScope.UnconfiguredProject, ProjectSystemContractProvider.Extension)]
+public interface IProjectItemContextMenuProvider
```

Comment:
I wonder whether we should copy the entire interface definition to the document as a general pattern in the document, because it is costly to maintain, and doesn't scale very well for the large set of interface.   Anyway, fine for this case...